vss-accessor/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+use ::prost::Message;
+use reqwest;
+use reqwest::Client;
+use std::error::Error;
+
+use crate::vss::{GetObjectRequest, GetObjectResponse, KeyValue, ListKeyVersionsRequest, ListKeyVersionsResponse, PutObjectRequest, PutObjectResponse};
+use crate::vss_error::VssError;
+
+mod vss_error;
+
+pub mod vss {
+    include!(concat!(env!("OUT_DIR"), "/org.vss.rs"));
+}
+
+pub struct VssAccessor {
+    base_url: String,
+    client: Client,
+}
+
+impl VssAccessor {
+    pub fn new(base_url: &str) -> Result<Self, Box<dyn Error>> {
+        let client = Client::new();
+        Ok(Self { base_url: String::from(base_url), client })
+    }
+
+    pub async fn get(&self, store: &str, key: &str) -> Result<GetObjectResponse, VssError> {
+        let url = format!("{}/getObject", self.base_url);
+
+        let request = GetObjectRequest { store_id: store.into(), key: key.to_string() };
```

Comment:
If you need ownership of method parameters, you should take them in by value instead of reference (i.e., `String` instead of `&str`). Likewise in other methods.

Reply:
Done 👍